test/core/security/local_security_connector_test.cc

```diff
@@ -0,0 +1,114 @@
+//
+//
+// Copyright 2024 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//
+
+#include <gtest/gtest.h>
+
+#include "src/core/lib/security/context/security_context.h"
+#include "src/core/tsi/transport_security.h"
+#include "test/core/test_util/test_config.h"
+
+namespace grpc_core {
+namespace testing {
+
+class LocalSecurityConnectorTest : public ::testing::Test {
+ protected:
+  LocalSecurityConnectorTest() {}
+};
+
+static absl::string_view me_get_local_address_unix(grpc_endpoint* /*ep*/) {
+  return "unix:";
+}
+
+static const grpc_endpoint_vtable vtable_unix = {nullptr,
+                                            nullptr,
+                                            nullptr,
+                                            nullptr,
+                                            nullptr,
+                                            nullptr,
+                                            nullptr,
+                                            me_get_local_address_unix,
+                                            nullptr,
+                                            nullptr};
+
+static absl::string_view me_get_local_address_local(grpc_endpoint* /*ep*/) {
+  return "ipv4:127.0.0.1:12667";
+}
+
+static const grpc_endpoint_vtable vtable_local = {nullptr,
+                                            nullptr,
+                                            nullptr,
+                                            nullptr,
+                                            nullptr,
+                                            nullptr,
+                                            nullptr,
+                                            me_get_local_address_local,
+                                            nullptr,
+                                            nullptr};
+
+static void check_tsi_security_level(grpc_local_connect_type connect_type,
```

Comment:
This test only checks the security level for the server credentials - can we have an analogous test for the channel credentials?